From: SeongJae Park <sj@kernel.org>
Date: Wed, 17 Dec 2025 14:17:52 -0800
Subject: [PATCH] (fixup) mm/damon/sysfs: cleanup sample primitives/ when
 filters/ setup fails

The function for setting internal directories of sample/ dir is not
cleaning up primitives/ directory when filter/ directory allocation is
failed.  Do that.

Reported-by: kernel test robot <lkp@intel.com>
Reported-by: Dan Carpenter <dan.carpenter@linaro.org>
Closes: https://lore.kernel.org/r/202512141528.AD9GSzHJ-lkp@intel.com/
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 4208fed2b8df..447fb4c6898f 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -917,7 +917,7 @@ static int damon_sysfs_sample_add_dirs(
 
 	filters = damon_sysfs_sample_filters_alloc();
 	if (!filters)
-		return -ENOMEM;
+		goto put_primitives_out;
 	err = kobject_init_and_add(&filters->kobj,
 			&damon_sysfs_sample_filters_ktype, &sample->kobj,
 			"filters");
-- 
2.47.3

